### PR TITLE
[F] Client changes to support authoring

### PR DESCRIPTION
**Pull request by 1aurend**
Renames `zd/authoring` to `feature/authoring`
From #3503

### CODE REVIEW

**Review by dananjohnson — CHANGES_REQUESTED** [review]
@1aurend see below for the notes I took while testing this. I know I flagged a number of things, but overall this looks really really good! Impressive work across the whole stack!! 👏🏻👏🏻👏🏻 

- I was able to create a new text without filling in any of the fields. At least the title should be required, right?
- The focus state for this button has insufficient contrast. Maybe we just use the default focus style rather than a custom one for this button class.
Screen Shot 2022-12-06 at 5 54 03 PM

- The modal for deleting a text as missing a value for l10n string `backend_entities.texts.modals.delete_text_message`
- I left the Section Link select blank when creating a TOC entry. I got an error as expected, but the entry title I'd inputted was wiped in the process. Is it possible to not reset the form if there's an error?
- Doing any combining with keyboard involving the first TOC entry (nesting _into_ it, or nesting it into another entry) looks a bit rough:
Screen Shot 2022-12-06 at 8 28 24 PM

- Obscure, but if you don't complete a drag event (like changing tabs) causes an error:
```js
List.js?f066:41 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'parentId')
    at onDragEnd (List.js?f066:41:1)
```

- I can actually reproduce the same error by dragging and releasing an item downwards out of the DnD area:

https://user-images.githubusercontent.com/12454203/206092395-99a336a9-b7dc-4013-8d44-5d37421a8f6e.mov

- The wrapping on this in Safari looks a bit unsightly:
Screen Shot 2022-12-06 at 8 45 27 PM

- A11y on the TOC interface looks pretty good. The semantics and announcements are almost identical to other DnD interfaces in Manifold, with the exception of the drag handle. In other places, it gets a `button` role and is described by a hidden div that explains how to use it. Here, it has no role, just an `aria-roledescription`, the value of which (in Voiceover at least) is not announced. Do we have any control over this? I'd try to make it match the other ones if at all possible.
- The other a11y thing to raise is the verbose description when items are combined. Is it possible to pass the entry title rather than the ID?
Screen Shot 2022-12-06 at 8 48 16 PM

- Mobile looks good! I just noticed one little issue:
Screen Shot 2022-12-06 at 8 56 08 PM

**Comment by 1aurend** [comment]
> @1aurend see below for the notes I took while testing this. I know I flagged a number of things, but overall this looks really really good! Impressive work across the whole stack!! 👏🏻👏🏻👏🏻

Thank you!

> 
>     * I was able to create a new text without filling in any of the fields. At least the title should be required, right?

I need to ask Zach and Alexa about this one. It's pretty clear in the api that title is intentionally not required, and any validation we add for texts created via this form shouldn't affect ingestion. I can think of a couple ways to do it, but not sure if either is a good idea. Same with text sections.

>     * The focus state for this button has insufficient contrast. Maybe we just use the default focus style rather than a custom one for this button class.

Sounds good. Will fix.

>     * The modal for deleting a text as missing a value for l10n string `backend_entities.texts.modals.delete_text_message`

This fix just went into v7/master in another PR.

>     * I left the Section Link select blank when creating a TOC entry. I got an error as expected, but the entry title I'd inputted was wiped in the process. Is it possible to not reset the form if there's an error?

Should be. Other forms don't. I'll look into it.
 
>     * Doing any combining with keyboard involving the first TOC entry (nesting _into_ it, or nesting it into another entry) looks a bit rough:

Huh? Yeah... there is a weird offset. I don't know if there's anything we can do about it without forking the tree library though. I'm going to add this one to my list of TOC things to polish in Linear.

>     * Obscure, but if you don't complete a drag event (like changing tabs) causes an error:
> 
>     * I can actually reproduce the same error by dragging and releasing an item downwards out of the DnD area: 

I don't think we can do anything about this other than maybe add the background color to the dropabble div while dragging like you suggested. Fwiw, these errors only occur in dev. The tree library includes an error boundary that catches them in prod. I tried to add my own boundary to customize handling, but the errors aren't propagated.
 
>     * The wrapping on this in Safari looks a bit unsightly:

Thanks for reminding me to use Safari 🙃 ... will fix.

>     * A11y on the TOC interface looks pretty good. The semantics and announcements are almost identical to other DnD interfaces in Manifold, with the exception of the drag handle. In other places, it gets a `button` role and is described by a hidden div that explains how to use it. Here, it has no role, just an `aria-roledescription`, the value of which (in Voiceover at least) is not announced. Do we have any control over this? I'd try to make it match the other ones if at all possible.

I don't know, but I can look at the other ones and try adding the role + describing div.

>     * The other a11y thing to raise is the verbose description when items are combined. Is it possible to pass the entry title rather than the ID?

I have no idea, but I suspect this would also required forking the tree library. I'll also add this one to Linear.
 
>     * Mobile looks good! I just noticed one little issue: 

Ooh yeah, need to truncate those.

File: client/src/global/components/drawer/Wrapper/index.js [hunk 1]
```diff
@@ -109,9 +109,9 @@ DrawerWrapper.propTypes = {
   lockScroll: PropTypes.string,
   entrySide: PropTypes.oneOf(["right", "left"]),
   context: PropTypes.oneOf(["backend", "frontend", "reader"]),
-  size: PropTypes.oneOf(["default", "wide", "flexible"]),
+  size: PropTypes.oneOf(["default", "wide", "flexible", "authoring"]),
```
[the remark this answers is not in the document]
Comment (1aurend) [reply client/src/global/components/drawer/Wrapper/index.js:112]: Will fix. I also need to connect with Lael at some point about the specific aspect ratio of this drawer.

File: client/src/global/components/form/BaseInput/index.js [hunk 2]
```diff
@@ -131,6 +133,10 @@ export class FormBaseInput extends PureComponent {
           type={this.props.inputType ?? this.props.type}
           placeholder={this.props.placeholder}
           onChange={this.props.onChange}
+          onKeyDown={e => {
+            if (this.props.onKeyDown)
+              this.props.onKeyDown(e, this.inputElement);
+          }}
```
[the remark this answers is not in the document]
Comment (1aurend) [reply client/src/global/components/form/BaseInput/index.js:139]: True. I did this so I could use the same handler function as I'm using for the create button without adding a conditional to it, but that could be just laziness.

**Review by dananjohnson — APPROVED** [review]

**Pull request merged**